rpi/default.nix

```diff
@@ -337,12 +205,43 @@ in
       kernelPackages = pkgs.linuxPackagesFor pkgs.rpi-kernels."${version}"."${board}";
       loader = {
         grub.enable = lib.mkDefault false;
-        initScript.enable = !cfg.uboot.enable;
-        generic-extlinux-compatible = {
-          enable = lib.mkDefault cfg.uboot.enable;
+
+        generic-extlinux-compatible.enable = false;
+        generic-extlinux-compatible-pi-loader = {
+          # extlinux-style boot is only used when uboot is enabled
+          # when uboot is disabled, use this module to put files into
+          # the boot partition as part of installBootloader
```

Comment:
how does this work with `direct-to-kernel`? Where do we install and update the kernel, initrd, params, etc in that case?

Reply:
`generic-extlinux-compatible-pi-loader` still installs the kernel and initrd in `/boot/nixos` - the same files as used for uboot, only the boot strategy is changed.

`cmdline.txt` is written to `/boot` in `config.generic-extlinux-compatible-pi-loader.extraCommandsAfter`